### DISCUSSION TITLE
Powerstream - set AC output value

### DISCUSSION THREAD

**Discussion by bogdancs92** [tolwi/hassio-ecoflow-cloud#130] [General]
Hi, 
I found this function : 
// Einstellen der Einspeiseleistung
function setAC(asn, Value) {
    if (!ConfigData.Regulation) {
        //return
    }
    let updatedMusterSetAC = musterSetAC;
    if (Value <= -1) {
        delete updatedMusterSetAC.item.meta;
        delete updatedMusterSetAC.item.ValByte;
    }
    else {
        updatedMusterSetAC.header.pdata.value = Value
        updatedMusterSetAC.header.dataLen = getVarintByteSize(Value)
    }
    updatedMusterSetAC.header.seq = Date.now()
    updatedMusterSetAC.header.deviceSn = asn
    //log(JSON.stringify(updatedMusterSetAC))
    setState(ConfigData.statesPrefix + '.app_' + mqttDaten.UserID + '_' + asn + '_thing_property_set.setAC', Value.toString(), true)
    SendProto(JSON.stringify(updatedMusterSetAC), '/app/' + mqttDaten.UserID + '/' + asn + '/thing/property/set');
}
[1692266227129-ecoflow-connector_v112.js.zip](https://github.com/tolwi/hassio-ecoflow-cloud/files/12388512/1692266227129-ecoflow-connector_v112.js.zip)

from the script provided on this page https://forum.iobroker.net/topic/66743/ecoflow-connector-script-zur-dynamischen-leistungsanpassung/2 for IOBrocker...

is it possible to adapt it ?

**Comment by mabeln68** [comment]
This would be a perfect solution for me too!

**Reply by Henschga** [reply]
same here
